RL-update-v2.2.0

ivukovicRL opened the pull request:
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

## Pull Request Checklist

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
- [ ] Limit your pull request to one application and submit multiple pull requests if necessary.
## Pull Request Type

#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.


## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

ivukovicRL committed:
v1.1.3 to v2.2.0

ivukovicRL committed:
Merge pull request #8 from ivukovicRL/rl-update2

v1.1.3 to v2.2.0

ivukovicRL committed:
Add tiscale view for test purposes

ivukovicRL committed:
Merge pull request #9 from ivukovicRL/rl-update2

Add tiscale view for test purposes

ivukovicRL committed:
Add required minimal data paths

ivukovicRL committed:
Merge pull request #10 from ivukovicRL/rl-update2

Add required minimal data paths

mjiyani-crest commented:
Hey @ivukovicRL, we will take care of these Static tests failures. If your other PR(https://github.com/splunk-soar-connectors/reversinglabs-tiscale/pull/4) is consisting the same changes, then you can close that PR.

Thanks,
Mit

ivukovicRL commented:
Hi @mjiyani-crest and thanks for the support. RL-update PR is deleted. We have a few questions regarding the PR.

There was an obsolete view **tiscale_view.py** which we deleted in this PR but it was still there causing errors during the checks (as if it was cached somewhere). So we added it again as an empty file in this commit https://github.com/splunk-soar-connectors/reversinglabs-tiscale/pull/5/commits/a0e11a458d6390e3c9fa1d15f5f631891c7d65b1 so the test could pass. Can you delete it for us? 

We also need to update our docs before the merge. Do you have some kind of a template for the docs? Also is the md enough or do we need to generate both html and md for the docs?

mjiyani-crest commented:
Hey @ivukovicRL, thanks for reverting back.

> There was an obsolete view **tiscale_view.py** which we deleted in this PR but it was still there causing errors during the checks (as if it was cached somewhere). So we added it again as an empty file in this commit [a0e11a4](https://github.com/splunk-soar-connectors/reversinglabs-tiscale/commit/a0e11a458d6390e3c9fa1d15f5f631891c7d65b1) so the test could pass. Can you delete it for us?

Once you will finish your changes, we will start working on this PR and take care of it.

> We also need to update our docs before the merge. Do you have some kind of template for the docs? Also is the md enough or do we need to generate both HTML and md for the docs?

Yes, you have to create/update an HTML file only from now onwards. You can add all the details which are necessary for the user to take care of while using the app.

dgopani-crest commented:
Hello @ivukovicRL, Are you done with the changes? 
Can we start further activities to release the app?

ivukovicRL commented:
> Hello @ivukovicRL, Are you done with the changes? Can we start further activities to release the app?

Hi @dgopani-crest, thanks for asking. We are almost done. There is is still one more deprecated function in our code, i will let you know when we fix this. 

Also we will soon open new PRs on https://github.com/splunk-soar-connectors/reversinglabs-ticloud and https://github.com/splunk-soar-connectors/reversinglabs-a1000

ivukovicRL committed:
Fix deprecated, update license and readme

ivukovicRL committed:
Merge pull request #13 from ivukovicRL/rl-update2

Fix deprecated, update license and readme

ivukovicRL committed:
Fix missing newline

ivukovicRL committed:
Merge pull request #14 from ivukovicRL/rl-update2

Fix missing newline

ivukovicRL committed:
Remove unused imports

ivukovicRL committed:
Merge pull request #15 from ivukovicRL/rl-update2

Remove unused imports

dgopani-crest commented:
> Hi @dgopani-crest, thanks for asking. We are almost done. There is still one more deprecated function in our code, I will let you know when we fix this.

Thanks for the quick response @ivukovicRL. Please let us know once all the changes are updated, we will start further release activities

ivukovicRL commented:
@dgopani-crest Is there a way to auto-generate a readme out of a config file (for actions outputs)? Also, do you have a preferred tool to convert a markdown to html? Is there a documentation describing how the documentation works? It seems that the docs for the app here 
![image](https://user-images.githubusercontent.com/71018129/149106146-91d76ae1-bb12-496e-91ab-4d9ee9110a0e.png)
are a combination of the markdown file and a config file. Is that correct?

dgopani-crest commented:
@ivukovicRL, the Documentation mentioned in the screenshot will be auto-generated by your readme.html and <app>.json file and looks like [this](https://my.phantom.us/5.1/docs/app_reference/phantom_reversinglabs). The above section of `Configuration Variables` is generated by readme.html and below that is generated from json file.

readme.md is also a combination of these 2 files, which will help users to know about an app on the GitHub repo. For README.md do not worry, we will take care of that and will create it from mentioned 2 files. So you can update all the required data in the readme.html & json file, it will automatically reflect on the Documentation page

Also, this MR shows "This branch has conflicts that must be resolved" can you please check?
<img width="842" alt="Screenshot 2022-01-12 at 4 17 50 PM" src="https://user-images.githubusercontent.com/64279254/149126249-edad7fcc-9ce9-4e21-ba98-24eeb5387e91.png">

ivukovicRL committed:
Merge branch 'splunk-soar-connectors-next' into rl-update2

ivukovicRL committed:
Merge branch 'rl-update2' of https://github.com/reversinglabs/reversinglabs-tiscale-splunk-soar into rl-update2

ivukovicRL commented:
@dgopani-crest It should be ok now.

dgopani-crest commented:
> @dgopani-crest It should be ok now.

Thanks @ivukovicRL, we will start further activities and release the app

ebhoir-crest committed:
Minor changes for handling exceptions

ebhoir-crest commented:
Hi @ivukovicRL, I am unable to push the changes to branch 'reversinglabs:rl-update2' with below error 'Permission to reversinglabs/reversinglabs-tiscale-splunk-soar.git denied to ebhoir-crest.' Could you please provide me the push access rights.

ivukovicRL commented:
Hi @ebhoir-crest you are added as a collaborator. Please let me know if you have any more issues.

pdesai-crest commented on tiscale_consts.py:
```diff
@@ -1,6 +1,6 @@
 # --
 #
-# Copyright (c) ReversingLabs Inc 2016-2019
+# Copyright (c) ReversingLabs Inc 2016Copyright (c) ReversingLabs Inc 2016-2022
```
Fix this license

ebhoir-crest replied:
fixed the license

ebhoir-crest committed:
Minor changes in license and example values

ebhoir-crest commented:
Static test failures are added to expected failures list and MR has been raised for the same. Once the expected failure MR is merged it will resolve the static tests.

ricardo-phantom commented on NOTICE:
```diff
@@ -1,5 +1,5 @@
 Splunk SOAR RL TitaniumScale Enterprise File Visibility
-Copyright (c) ReversingLabs Inc 2016-2019
+Copyright (c) ReversingLabs Inc 2016-2021
```
2022

ebhoir-crest replied:
Thanks @ricardo-phantom , updated the year

ebhoir-crest committed:
Updating copyright year

rgil-splunk commented on reversinglabs_view.py:
```diff
@@ -1,40 +1,136 @@
+    for category, values in hunting_meta_stats.items():
+        readable_category = ' '.join(category.split('_'))
+        stats_fields['categories'].append(readable_category)
+        for key, value in list(values.items()):
```
```suggestion
        for key, value in values.items():
```

rgil-splunk commented on tiscale_connector.py:
```diff
@@ -184,7 +183,7 @@ def _parse_report_status_msg(self, response, action_result, data):
         response1['tiscale_link'] = "{0}{1}".format(
             self._base_url, "?q=" + data['hash'])
-        print "TISCALE link:" + response1['tiscale_link']
+        print("TISCALE link:" + str(response1['tiscale_link']))
```
can we either make this a `debug_print` or remove it? also, use string formatting instead of concatenation if it's kept

rgil-splunk commented on tiscale_connector.py:
```diff
@@ -251,18 +250,20 @@ def _make_rest_call(
                         phantom.APP_ERROR,
-                        "REST POST Api to server failed " +
-                        str(e),
+                        "REST POST Api to server failed " + str(e),
```
```suggestion
                        "REST POST API to server failed {}".format(e),
```

rgil-splunk commented on tiscale_connector.py:
```diff
@@ -251,18 +250,20 @@ def _make_rest_call(
             try:
                 if(TISCALE_JSON_API_KEY in config):
```
```suggestion
                if TISCALE_JSON_API_KEY in config:
```

rgil-splunk commented on tiscale_connector.py:
```diff
@@ -273,18 +274,17 @@ def _make_rest_call(
                 if(TISCALE_JSON_API_KEY in config):
```
```suggestion
                if TISCALE_JSON_API_KEY in config:
```

rgil-splunk commented on tiscale_connector.py:
```diff
@@ -251,18 +250,20 @@ def _make_rest_call(
-                                      'Authorization': 'Token %s' % config[TISCALE_JSON_API_KEY]})
+                    r = requests.post(url, files=files, timeout=10, headers={
+                                      'Authorization': 'Token %s' % config[TISCALE_JSON_API_KEY],
+                                      'User-Agent': 'ReversingLabs Phantom TiScale v2.2'})
```
can we make the user agent string a constant? we seem to reuse it a couple of times

rgil-splunk commented on tiscale_connector.py:
```diff
@@ -273,18 +274,17 @@ def _make_rest_call(
-                        str(e) +
-                        " url: " +
-                        url,
+                        "REST GET Api to server failed " + str(e) + " url: " + url,
```
```suggestion
                        "REST GET API to server failed {} url: {}".format(e, url),
```

rgil-splunk commented on tiscale_connector.py:
```diff
@@ -312,19 +312,21 @@ def _make_rest_call(
+                return (action_result.set_status(phantom.APP_ERROR,
+                        f'Unable to get Vault item details. Error Details: {msg}'),
+                        None)
+            file_data = list(files_array)[0]
```
```suggestion
            file_data = next(iter(files_array))
```
to avoid generating the whole list

rgil-splunk commented on tiscale_connector.py:
```diff
@@ -632,101 +513,18 @@ def validate_parameters(self, param):
-        vault_id = param['vault_id']
-        ret_val, sha256 = self._get_vault_file_sha256(vault_id, action_result)
-
         if (phantom.is_fail(ret_val)):
```
```suggestion
        if phantom.is_fail(ret_val):
```

rgil-splunk commented on tiscale_connector.py:
```diff
@@ -763,6 +571,51 @@ def _detonate_file(self, param):
+        hunting_report_vault_id = parameters.get(TISCALE_JSON_HUNTING_STATE)
+        if hunting_report_vault_id:
+            success, msg, files_array = ph_rules.vault_info(vault_id=hunting_report_vault_id)
+            file_data = list(files_array)[0]
```
```suggestion
            file_data = next(iter(files_array))
```

rgil-splunk commented on tiscale_consts.py:
```diff
@@ -29,7 +33,8 @@
 TISCALE_SLEEP_SECS = 10
 TISCALE_MSG_REPORT_PENDING = "Report Pending"
-TISCALE_MSG_MAX_POLLS_REACHED = "Reached max polling attempts. Please use the MD5 or Sha256 of the file as a parameter to <b>get report</b> to query the report status."
+TISCALE_MSG_MAX_POLLS_REACHED = """Reached max polling attempts.
+Please use the MD5 or Sha256 of the file as a parameter to <b>get report</b> to query the report status."""
```
```suggestion
TISCALE_MSG_MAX_POLLS_REACHED = "Reached max polling attempts. " \
    "Please use the MD5 or Sha256 of the file as a parameter to <b>get report</b> to query the report status."
```
a multiline string would preserve the literal spacing in the printed string

ebhoir-crest committed:
Minor changes for coding standards

ebhoir-crest committed:
added new constant

ebhoir-crest committed:
updating new constant

ebhoir-crest committed:
minor changes and removed unsed files

ebhoir-crest commented:
Hi @ivukovicRL can you please suggest for the points to be added in unreleased.md for current app changes

ebhoir-crest commented:
Hi @ivukovicRL can you please suggest points for unreleased.md ?

dgopani-crest commented:
Hi @ivukovicRL, currently we have added below one point for release notes.  we are seeing backward compatibility changes in the action parameters so other than this point do we need to add any other release notes point?

- Compatibility changes for Python 3 support 

Also, we have made a few best practices changes in the PR. Please have a look into it and let us know in case of any queries.

ivukovicRL commented:
@dgopani-crest  We agree with the changes.

ebhoir-crest committed:
Minor changes in constants and release notes

ebhoir-crest committed:
Updating minimum phantom version

ebhoir-crest committed:
Updating minimum phantom version and contains parameter

ebhoir-crest commented:
Static test fails as `min_phantom_version` will be `5.0.0`

ivukovicRL commented:
Hi @dgopani-crest is there anything else that we need to do before the release?

Edit: It is released, thanks!

dgopani-crest commented:
> Hi @dgopani-crest is there anything else that we need to do before the release?

@ivukovicRL, the app is released and available on the [splunkbase](https://splunkbase.splunk.com/app/6069/) and [my.phantom.us](https://my.phantom.us/5.0/apps/)
